core/types/header_ext_test.go

```diff
@@ -242,3 +242,55 @@ func TestHeaderSerializable_updates(t *testing.T) {
 		assert.Equal(t, wantedExtras, extras)
 	})
 }
+
+func TestHeaderExtraPostCopy(t *testing.T) {
+	t.Parallel()
+
+	t.Run("empty_extra", func(t *testing.T) {
+		t.Parallel()
+
+		got := &ethtypes.Header{
+			ParentHash: common.Hash{1},
+		}
+		extra := &HeaderExtra{}
+
+		extra.PostCopy(got)
+
+		want := &ethtypes.Header{
+			ParentHash: common.Hash{1},
+		}
+		wantExtra := &HeaderExtra{}
+		_ = WithHeaderExtra(want, wantExtra)
```

Comment:
```suggestion
		wantExtra := &HeaderExtra{}
		want := WithHeaderExtra(
			&ethtypes.Header{
				ParentHash: common.Hash{1},
				Number: big.NewInt(0),
				Difficulty: big.NewInt(0),
			},
			wantExtra,
		)
```

This addresses:

1. The need for the explicit zeroes due to using `CopyHeader()`.
2. Refactoring so the `WithHeaderExtra()` makes more sense.